### PR TITLE
OpenCL: Add rev support for unary log* functions

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

Add rev OpenCL support for: log, log10, log1m, log1p, log2, logit, log1m_exp, log1p_exp, log_inv_logit, log1m_inv_logit

log1m and log_inv_logit also required prim implementations

## Tests

All new functions are tested.

## Side Effects

/

## Release notes

OpenCL: Added /rev OpenCL support for: log, log10, log1m, log1p, log2, logit, log1m_exp, log1p_exp, log_inv_logit, log1m_inv_logit

## Checklist

- [x] Math issue #1854 

- [x] Copyright holder: Rok Češnovar

### CODE REVIEW

File: stan/math/opencl/rev/log1p.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * @param A argument
+ * @return Elementwise `log1p()` of the input.
+ */
+inline var_value<matrix_cl<double>> log1p(
+    const var_value<matrix_cl<double>>& A) {
+  var_value<matrix_cl<double>> res = log1p(A.val());
+
+  reverse_pass_callback([A, res]() mutable {
+    A.adj() = A.adj() + elt_divide(res.adj(), 1 + A.val());
```
Comment (t4c1) [inline stan/math/opencl/rev/log1p.hpp:23]:
```suggestion
    A.adj() = A.adj() + elt_divide(res.adj(), 1.0 + A.val());
```
For kernel generator expressions we can't rely on the compiler to promote constants to appropriate type. So the type conversion would happen in the kernel.

File: stan/math/prim/fun/log10.hpp [hunk 2]
```diff
@@ -48,8 +50,10 @@ inline auto log10(const Container& x) {
  * @param x Container
  * @return Log base-10 of each variable in the container.
  */
-template <typename Container,
-          require_container_st<std::is_arithmetic, Container>* = nullptr>
+template <
+    typename Container,
+    require_container_st<std::is_arithmetic, Container>* = nullptr,
+    require_not_nonscalar_prim_or_rev_kernel_expression_t<Container>* = nullptr>
```
Comment (t4c1) [inline stan/math/prim/fun/log10.hpp:56]: I don't think the addition of this require is needed.

File: stan/math/prim/fun/logit.hpp [hunk 3]
```diff
@@ -100,8 +102,10 @@ inline auto logit(const Container& x) {
  * Note: The return must be evaluated otherwise the Ref object falls out
  * of scope
  */
-template <typename Container,
-          require_container_st<std::is_arithmetic, Container>* = nullptr>
+template <
+    typename Container,
+    require_container_st<std::is_arithmetic, Container>* = nullptr,
+    require_not_nonscalar_prim_or_rev_kernel_expression_t<Container>* = nullptr>
```
Comment (t4c1) [inline stan/math/prim/fun/logit.hpp:108]: Same here. I think we don't need this require.

File: test/unit/math/opencl/kernel_generator/elt_function_cl_test.cpp [hunk 4]
```diff
@@ -50,6 +50,7 @@ TEST_UNARY_FUNCTION(log)
 TEST_UNARY_FUNCTION(log2)
 TEST_UNARY_FUNCTION(log10)
 TEST_UNARY_FUNCTION(log1p)
+TEST_UNARY_FUNCTION(log1m)
```
Comment (t4c1) [inline test/unit/math/opencl/kernel_generator/elt_function_cl_test.cpp:53]: What about `log_inv_logit` test?

**Review by t4c1 — APPROVED** [review]

**Review by t4c1 — CHANGES_REQUESTED** [review]
Whoops, that was a misclick. I meant to request changes

File: stan/math/prim/fun/log10.hpp (hunk 2 above)
Comment (rok-cesnovar) [reply stan/math/prim/fun/log10.hpp:56]: True. Removed.

**Review by t4c1 — APPROVED** [review]

**Pull request merged**